test/unit/db/repos/migrations.js

```diff
@@ -0,0 +1,355 @@
+				const account = fixtures.accounts.Account();
+				yield db.accounts.insert(account);
+				const data = fixtures.accounts.Dependent({
+					accountId: account.address,
+				});
+
+				yield db.query(
+					db.$config.pgp.helpers.insert(data, null, {
+						table: 'mem_accounts2multisignatures',
+					})
+				);
+				yield db.migrations.applyRuntime();
+
+				const result = yield db.query(
+					'SELECT * FROM mem_accounts2u_multisignatures'
+				);
+
+				expect(result).to.have.lengthOf(1);
+				return expect(result[0]).to.be.eql(data);
+			});
+		});
+
+		describe('createMemoryTables()', () => {
+			it('should use the correct SQL while in transaction context', function*() {
+				return yield db.tx(function*(t) {
+					sinonSandbox.spy(t, 'none');
+
+					yield t.migrations.createMemoryTables();
+
+					return expect(t.none.firstCall.args[0]).to.be.eql(
+						migrationsSQL.memoryTables
+					);
+				});
+			});
+
+			it('should start a transaction context if no transaction exists', function*() {
+				sinonSandbox.spy(db, 'tx');
+
+				yield db.migrations.createMemoryTables();
+
+				expect(db.tx.firstCall.args[0]).to.be.eql('createMemoryTables');
+				return expect(db.tx.firstCall.args[1]).to.be.a('function');
+			});
+		});
+
+		describe('readPending()', () => {
+			let files;
+			let fileIds;
+
+			beforeEach(function*() {
+				files = yield fs.readdir(path.join(sqlRoot, 'migrations', 'updates'));
+				fileIds = files.map(f => f.match(/(\d+)_(.+).sql/)[1]).sort();
+				return fileIds;
+			});
+
+			it('should resolve with list of pending files if there exists any', function*() {
+				const pending = (yield db.migrations.readPending(fileIds[0])).map(
+					f => f.id
+				);
+
+				fileIds.splice(0, 1);
+				return expect(pending).to.be.eql(fileIds);
+			});
+
+			it('should resolve with empty array if there is no pending migration', function*() {
+				const pending = yield db.migrations.readPending(
+					fileIds[fileIds.length - 1]
+				);
+
+				return expect(pending).to.be.empty;
+			});
+
+			it('should resolve the list in correct format', function*() {
+				const pending = yield db.migrations.readPending(fileIds[0]);
+
+				expect(pending).to.be.an('array');
+				expect(pending[0]).to.have.all.keys('id', 'name', 'path', 'file');
+				return expect(pending[0].file).to.be.instanceOf(
+					db.$config.pgp.QueryFile
+				);
+			});
+		});
+
+		describe('applyAll()', () => {
+			let files;
+			let fileIds;
+			let fileNames;
+			let updates;
+
+			beforeEach(() => {
+				return fs
+					.readdir(path.join(sqlRoot, 'migrations', 'updates'))
+					.then(result => {
+						files = result;
+						fileIds = files.map(f => f.match(/(\d+)_(.+).sql/)[1]).sort();
+						fileNames = files.map(f => f.match(/(\d+)_(.+).sql/)[2]);
+
+						return db.migrations.readPending(fileIds[0]);
+					})
+					.then(_updates => {
+						updates = _updates;
+					});
+			});
+
+			it('should apply all pending migrations in a transaction', () => {
```

Comment:
`in a transaction`, mean all migrations in single transaction or all migrations in separate transactions (one per migration)? specify please

Reply:
In one transaction as main and nested save points for each individual migration.